File: Resueltos_por_competidor/Juan_Martinez/RoadToZero.cpp
/*
 * Autor: Juan Martinez
 * Problema: Road To Zero (1342A)
 * Juez online: Codeforces
 * Veredicto: Accepted
 * Url: https://codeforces.com/problemset/problem/1342/A
 * Difficulty: 1000
 */
#include <bits/stdc++.h>
using namespace std;

typedef long long int ll;

int main() {
	ll t, x, y, a, b, dif, res; cin>>t;
    while(t--) {
        cin>>x>>y;
        cin>>a>>b;

        res = 0;
        dif = abs(x-y);
        res += (dif*a);
        if((2*a) > b) res += (b*min(x,y));
        else res += ((2*a)*min(x,y));
        cout<<res<<endl;
    }
}

File: Resueltos_por_competidor/Martin_Peña/RoadToZero.java
/*
* Autor: pgms1103-afk
* Problema: A. Road To Zero
* Juez Online: Codeforces
* Veredicto: Accepted
* URL: https://codeforces.com/problemset/problem/1342/A
*/
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class RoadToZero {

	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	public static void main(String[] args) throws NumberFormatException, IOException {
		
		int t = Integer.parseInt(br.readLine());
		for (int i = 0; i < t; i++) {
			
			String datos[] = br.readLine().split(" ");
			int x = Integer.parseInt(datos[0]);
			int y = Integer.parseInt(datos[1]);
			
			String datos2[] = br.readLine().split(" ");
			long a = Long.parseLong(datos2[0]);
			long b = Long.parseLong(datos2[1]);
			
			
			long max = Math.max(x, y);
			long min = Math.min(x, y);
			
			long respuesta1 = (max+min) * a;
			long respuesta2 = (max - min) * a + min*b;
					
//			while(max != 0 && min != 0) {
//				respuesta += b;
//				max--;
//				min--;
//			}
			System.out.println(Math.min(respuesta1, respuesta2));
			
		}

	}

}
